compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -3092,9 +3324,55 @@ abstract class GenJSCode extends plugins.PluginComponent
       val receiver = genExpr(receiver0)
       val args = genPrimitiveJSArgs(sym, args0)
 
+      genJSCallGeneric(sym, receiver, args, isStat)
+    }
+
+    private def genJSSuperCall(tree: Apply, isStat: Boolean): js.Tree = {
+      implicit val pos = tree.pos
+      val Apply(fun @ Select(sup @ Super(_, mix), _), args) = tree
+      val sym = fun.symbol
+
+      val genReceiver = genThis()(sup.pos)
+      lazy val genScalaArgs = genActualArgs(sym, args)
+      lazy val genJSArgs = genPrimitiveJSArgs(sym, args)
+
+      if (sym.owner == ObjectClass) {
+        // Normal call anyway
```

Comment:
Er no, it isn't.

That's a call like `super.hashCode()` which resolves to Object's `hashCode` implementation. But we can't really call `$c_jl_Object.prototype.hashCode__I.call(this)` here because `this` is *not* an instance of `jl.Object` at this point. However, it is perfectly valid to generate `ApplyMethod(this, "hashCode")` because, `this` being of type `AnyType`, this is rerouted to the helper `$objectHashCode`, which returns 42/identity hash code for the argument.